### PR TITLE
frontend/feat: #1054 In-App chat enhancement

### LINKED ISSUE #1054: In-App chat enhancement
**Is your feature request related to a problem? Please describe.**
Enhancing In-App chat feature

**Describe the solution you'd like**
Adding full screen chat, overlay for messages in driver, better in-app notifications for customer

**Describe alternatives you've considered**

### PR DESCRIPTION
**Pull request by PraveenGongada**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Added intelligent suggestions and new in-app notification.

Closes #1054

### CODE REVIEW

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/MessageOverlay.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        }
+
+        if (messageSheetHeader != null) {
+            messageSheetHeader.setOnClickListener(this);
+        }
+
+        suggestions = getDefaultSuggestions(context);
+
+        if(suggestions != null) {
+            System.out.println("BATMAN");
```
Comment (ashishsingh101) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/MessageOverlay.java:80]: remove batman

File: Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/MessageOverlay.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+class Suggestions {
+    String s1;
+    String s2;
+    String s3;
+    public Suggestions(String s1, String s2, String s3) {
+        this.s1 = s1;
+        this.s2 = s2;
+        this.s3 = s3;
+    }
+}
```
Comment (ashishsingh101) [inline Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/MessageOverlay.java:153]: no need for this class and function, instead you can use JSON Object or array

File: Frontend/ui-common/src/Engineering/Helpers/JBridge.purs [hunk 3]
```diff
@@ -374,4 +382,329 @@ getWidthFromPercent percent =
...
+
+  cis1BP : {key : "cis1BP", value : {en_us : "Are you starting ?", ta_in : "நீங்கள் தொடங்குகிறீர்களா?", kn_in : "ನೀವು ಪ್ರಾರಂಭಿಸುತ್ತಿದ್ದೀರಾ?", hi_in : "आप आ रहे है क्या? ", ml_in : "താങ്കൾ പുറപ്പെടുകയാണോ?", bn_in : "আপনি কি আসছেন ?" }},
+  cis2BP : {key : "cis2BP", value : {en_us : "Urgent, come soon", ta_in : "அவசரம், சீக்கிரம் வாருங்கள்", kn_in : "ಅರ್ಜೆಂಟ್, ಬೇಗ ಬನ್ನಿ", hi_in : "अर्जेंट है प्लीज जड़ली आइये", ml_in : "അത്യാവശ്യമാണ്, വേഗം വരൂ", bn_in : "জরুরী আছে, দয়া করে তাড়াতাড়ি আসুন" }},
+  cis3BP : {key : "cis3BP", value : {en_us : "Waiting at Pick-up", ta_in : "பிக்-அப்பில்  காத்திருக்கிறேன்", kn_in : "ಪಿಕ್-ಅಪ್‌ನಲ್ಲಿ ಕಾಯುತ್ತಿದ್ದೇನೆ", hi_in : "मैं लोकेशन पे आपके लिए रुका हूँ", ml_in : "ഞാൻ പിക്ക്-അപ്പിൽ കാത്തിരിക്കുകയാണ്", bn_in : "আমি পিক-আপ স্থানে অপেক্ষা করছি" }},
+  cis4BP : {key : "cis4BP", value : {en_us : "Call me once you reach", ta_in : "சேர்ந்தவுடன் கால் செய்யவும்", kn_in : "ನೀವು ತಲುಪಿದ ನಂತರ ಕರೆ ಮಾಡಿ", hi_in : "पहुँच जाओगे तब मुझे एक बार कॉल कर देना", ml_in : "താങ്കൾ എത്തിക്കഴിയുമ്പോൾ എന്നെ വിളിക്കുക", bn_in : "পৌঁছে গেলে আমাকে ফোন করুন" }},
+
```
Comment (ashishsingh101) [inline Frontend/ui-common/src/Engineering/Helpers/JBridge.purs:413]: you can move these to different file

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

**Review by MercyQueen — APPROVED** [review]

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 4]
```diff
@@ -265,6 +266,7 @@ supportButton push state =
   , height WRAP_CONTENT
   , orientation VERTICAL
   , gravity CENTER
+  , visibility if ((state.props.currentStage == RideAccepted) || (state.props.currentStage == RideStarted) || (state.props.currentStage == ChatWithDriver)) && (not state.props.showChatNotification) then VISIBLE else GONE
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:269]: @PraveenGongada can we create a function which would return this value as the same condition is being used at many places.

Comment (PraveenGongada) [reply Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:269]: Done

**Review by ashishsingh101 — APPROVED** [review]

**Pull request merged**